frontend/result.go

```diff
@@ -8,6 +8,7 @@ type Result struct {
 	Ref      solver.ResultProxy
 	Refs     map[string]solver.ResultProxy
 	Metadata map[string][]byte
+	Attrs    map[string]string
```

Comment:
I think we could avoid adding this. The `Metadata` could already contain the attrs sent by the frontend and we can just add to the same value. This is similar to the sources handling. It's a bit hackier as the value is `[]byte` but you can put the actual implementation to `buildinfo` package like the sources is done.